### PR TITLE
Towards full generic genotypes.

### PR DESCRIPTION
**Pull request by cseed**
Removed GenotypeStream.
Removed isLinearScale from VSM.
Removed write_partitioning.
{Variant, Sample}QC work over GDS.

Depends on https://github.com/hail-is/hail/pull/2039

### CODE REVIEW

File: src/main/scala/is/hail/variant/VariantSampleMatrix.scala [hunk 1]
```diff
@@ -1995,6 +1917,7 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
       fatal("found one or more type check errors")
   }
 
+
```
Comment (jigold) [inline src/main/scala/is/hail/variant/VariantSampleMatrix.scala]: delete

File: src/main/scala/is/hail/variant/VariantSampleMatrix.scala [hunk 2]
```diff
@@ -2261,10 +2158,7 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
       StructField("variant", vSignature.schema, nullable = false),
       StructField("annotations", vaSignature.schema),
       StructField("gs",
-        if (vSignature != TVariant || genotypeSignature != TGenotype || parquetGenotypes)
-          ArrayType(genotypeSignature.schema, containsNull = false)
-        else
-          GenotypeStream.schema,
+        ArrayType(genotypeSignature.schema, containsNull = false),
```
Comment (jigold) [inline src/main/scala/is/hail/variant/VariantSampleMatrix.scala]: parquetGenotypes not used anymore -- remove from function signature

File: src/test/scala/is/hail/methods/FilterSuite.scala [hunk 3]
```diff
@@ -40,7 +40,7 @@ class FilterSuite extends SparkSuite {
 
     assert(sQcVds.filterSamplesExpr("sa.qc.nCalled == 337").nSamples == 17)
 
-    assert(sQcVds.filterSamplesExpr("sa.qc.dpMean > 60").nSamples == 7)
+    assert(sQcVds.filterSamplesExpr("sa.qc.dpMean > 60").nSamples == 6)
```
Comment (jigold) [inline src/test/scala/is/hail/methods/FilterSuite.scala]: Double check 6 should be the new correct value and this isn't an artifact of having the wrong type in the GenotypeExtractor.

File: src/main/scala/is/hail/variant/Genotype.scala [hunk 4]
```diff
@@ -176,6 +176,32 @@ class RowGenotype(r: Row) extends Genotype {
...
+      val dpq = if (dpt == TArray(TInt)) dpq0 else (a: Any) => null
+
+      val (gqt, gqq0) = t.queryTyped("GQ")
+      val gqq = if (gqt == TInt) gqq0 else (a: Any) => null
+
+      val (plt, plq0) = t.queryTyped("PL")
+      val plq = if (plt == TInt) plq0 else (a: Any) => null
+
+      (a: Any) =>
+        new RowGenotype(Row(gtq(a), adq(a), dpq(a), gqq(0), plq(0)))
```
Comment (jigold) [inline src/main/scala/is/hail/variant/Genotype.scala]: RowGenotype will fail if someone calls `_fakeRef` or `_isLinearScale`. Need to define 6th and 7th element in row.

File: src/test/scala/is/hail/variant/vsm/VSMSuite.scala [hunk 5]
```diff
@@ -183,7 +183,7 @@ class VSMSuite extends SparkSuite {
   @Test def testWriteParquetRead() {
     val p = forAll(VariantSampleMatrix.gen(hc, VSMSubgen.random)) { vds =>
       val f = tmpDir.createTempFile(extension = "vds")
-      vds.write(f, parquetGenotypes = true)
+      vds.write(f)
```
Comment (jigold) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: Is this test still needed?

Comment (cseed) [inline src/test/scala/is/hail/variant/vsm/VSMSuite.scala]: No.  Removed.

**Pull request merged**